mysticeti-core/src/consensus/universal_committer.rs

```diff
@@ -26,20 +27,29 @@ impl UniversalCommitter {
     /// Try to commit part of the dag. This function is idempotent and returns a list of
     /// ordered decided leaders.
     #[tracing::instrument(skip_all, fields(last_decided = %last_decided))]
-    pub fn try_commit(&self, last_decided: BlockReference) -> Vec<LeaderStatus> {
+    pub fn try_commit(&self, last_decided: AuthorityRound) -> Vec<LeaderStatus> {
         let highest_known_round = self.block_store.highest_round();
         // let last_decided_round = max(last_decided.round(), 1); // Skip genesis.
         let last_decided_round = last_decided.round();
         let last_decided_round_authority = (last_decided.round(), last_decided.authority);
 
         // Try to decide as many leaders as possible, starting with the highest round.
         let mut leaders = VecDeque::new();
-        for round in (last_decided_round..=highest_known_round).rev() {
+        'outer: for round in (last_decided_round..=highest_known_round.saturating_sub(2)).rev() {
```

Comment:
should we provide more comments about `saturating_sub(2)`?

Reply:
Namely that we can only direct-decide a block when we have its decision round, and there cannot (yet) be an indirect-decision over a block that has no direct-committed children (which takes 5 extra rounds)

Reply:
Good point!